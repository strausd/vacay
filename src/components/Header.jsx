import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = () => {
    return (
        <div>
            <ul>
                <li><NavLink exact to="/" activeClassName="active">Home</NavLink></li>
                <li><NavLink to="/about" activeClassName="active">About</NavLink></li>
            </ul>
        </div>
    );
};

export default Header;