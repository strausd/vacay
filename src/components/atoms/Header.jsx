import React from 'react';
import { connect } from 'react-redux';
import { NavLink, withRouter } from 'react-router-dom';

import UserMenu from './UserMenu';


class Header extends React.Component {

    render() {
        return (
            <div>
                <div className="header">
                    <div className="header__content">
                        <ul>
                            <li><NavLink exact to="/" activeClassName="active">Home</NavLink></li>
                            <li><NavLink exact to="/pricing" activeClassName="active">Pricing</NavLink></li>
                            <li><NavLink exact to="/contact" activeClassName="active">Contact</NavLink></li>
                            {this.props.isAuthenticated ? null : <li><NavLink exact to="/signup" activeClassName="active">Signup</NavLink></li>}
                            {this.props.isAuthenticated ? null : <li><NavLink exact to="/login" activeClassName="active">Login</NavLink></li>}
                            {!this.props.isAuthenticated ? null : <li><NavLink exact to="/dashboard" activeClassName="active">Dashboard</NavLink></li>}
                        </ul>
                        <UserMenu />
                    </div>
                    <div className="header__transition"></div>
                </div>
            </div>
        );
    }
};

const mapStateToProps = (state) => {
    return {
        isAuthenticated: state.auth.jwt ? true : false
    };
};

export default withRouter(connect(mapStateToProps)(Header));