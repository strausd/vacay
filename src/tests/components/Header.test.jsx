import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../components/Header';


test('should render Header component properly', () => {
    const wrapper = shallow(<Header />);
    
    expect(wrapper).toMatchSnapshot();
});