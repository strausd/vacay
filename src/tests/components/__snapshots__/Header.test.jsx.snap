// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            exact={true}
            to="/"
          >
            Home
          </NavLink>
        </li>
        <li>
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/projects"
          >
            Projects
          </NavLink>
        </li>
        <li>
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/resume"
          >
            Resume
          </NavLink>
        </li>
        <li>
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/about"
          >
            About
          </NavLink>
        </li>
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              to="/"
            >
              Home
            </NavLink>
          </li>,
          <li>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/projects"
            >
              Projects
            </NavLink>
          </li>,
          <li>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/resume"
            >
              Resume
            </NavLink>
          </li>,
          <li>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/about"
            >
              About
            </NavLink>
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              to="/"
            >
              Home
            </NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Home",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/projects"
            >
              Projects
            </NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Projects",
              "to": "/projects",
            },
            "ref": null,
            "rendered": "Projects",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/resume"
            >
              Resume
            </NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Resume",
              "to": "/resume",
            },
            "ref": null,
            "rendered": "Resume",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/about"
            >
              About
            </NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "About",
              "to": "/about",
            },
            "ref": null,
            "rendered": "About",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              to="/"
            >
              Home
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/projects"
            >
              Projects
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/resume"
            >
              Resume
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/about"
            >
              About
            </NavLink>
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                exact={true}
                to="/"
              >
                Home
              </NavLink>
            </li>,
            <li>
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/projects"
              >
                Projects
              </NavLink>
            </li>,
            <li>
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/resume"
              >
                Resume
              </NavLink>
            </li>,
            <li>
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/about"
              >
                About
              </NavLink>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                exact={true}
                to="/"
              >
                Home
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Home",
                "exact": true,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/projects"
              >
                Projects
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Projects",
                "to": "/projects",
              },
              "ref": null,
              "rendered": "Projects",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/resume"
              >
                Resume
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Resume",
                "to": "/resume",
              },
              "ref": null,
              "rendered": "Resume",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/about"
              >
                About
              </NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "About",
                "to": "/about",
              },
              "ref": null,
              "rendered": "About",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
